kernel/shared/src/main/scala/monocle/Iso.scala

```diff
@@ -2,37 +2,13 @@ package monocle
 
 import monocle.function.Reverse
 
-abstract class Iso[A, B] extends Lens[A, B] with Prism[A, B] { self =>
-  override def modify(f: B => B): A => A =
-    from => reverseGet(get(from))
-
-  override def asTarget[C](implicit ev: B =:= C): Iso[A, C] =
-    asInstanceOf[Iso[A, C]]
-
-  def compose[C](other: Iso[B, C]): Iso[A, C] = new Iso[A, C] {
-    def get(from: A): C      = other.get(self.get(from))
-    def reverseGet(to: C): A = self.reverseGet(other.reverseGet(to))
-  }
-  override def compose[C](other: Lens[B, C]): Lens[A, C] = new Lens[A, C] {
-    def get(from: A): C    = other.get(self.get(from))
-    def set(to: C): A => A = from => self.reverseGet(other.set(to)(self.get(from)))
-  }
-  override def compose[C](other: Prism[B, C]): Prism[A, C] = new Prism[A, C] {
-    def getOption(from: A): Option[C] = other.getOption(self.get(from))
-    def reverseGet(to: C): A          = self.reverseGet(other.reverseGet(to))
-  }
-}
-
 object Iso {
-  def apply[A, B](_get: A => B)(_reverseGet: B => A): Iso[A, B] =
-    new Iso[A, B] {
-      def get(from: A): B      = _get(from)
-      def reverseGet(to: B): A = _reverseGet(to)
-    }
+  def apply[S, A](_get: S => A)(_reverseGet: A => S): Iso[S, A] =
+    PIso[S, S, A, A](_get)(_reverseGet)
 
-  def reverse[A, B](implicit ev: Reverse.Aux[A, B]): Iso[A, B] =
+  def reverse[S, A](implicit ev: Reverse.Aux[S, A]): Iso[S, A] =
```

Comment:
I think we standardized on using `A, B, ..` for type params rather than `S, A`

Reply:
We might need to revisit that if we do go for polymorphic optics. We could have polymorphic optics with A1, A2, B1, B2, e.g. `PLens[A1, A2, B1, B2]` and `Lens[A, B]`